src/wp-includes/option.php

```diff
@@ -364,6 +364,157 @@ function get_options( $options ) {
 	return $result;
 }
 
+/**
+ * Sets the autoload value for an option in the database.
+ *
+ * Autoloading too many options can lead to performance problems, especially if the options are not frequently used.
+ * This function allows modifying the autoload value for an option without changing the actual option value. This is
+ * for example recommended for plugin activation and deactivation hooks, to ensure any options exclusively used by the
+ * plugin which are generally autoloaded can be set to not autoload when the plugin is inactive.
+ *
+ * Alternatively, {@see wp_set_options_autoload()} can be used to set the autoload value for multiple options at once.
+ *
+ * @since 6.4.0
+ *
+ * @global wpdb $wpdb WordPress database abstraction object.
+ *
+ * @param string      $option   Name of the option. Expected to not be SQL-escaped.
+ * @param string|bool $autoload Autoload value to control whether to load the option when WordPress starts up.
+ *                              Accepts 'yes'|true to enable or 'no'|false to disable.
+ * @return bool True if the autoload value was modified, false otherwise.
+ */
+function wp_set_option_autoload( $option, $autoload ) {
+	global $wpdb;
+
+	$autoload = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+
+	/*
+	 * Determine the current autoload value for the option.
+	 * If the value is empty, it means the option does not exist in the database.
+	 * If the old and new value are the same, no need to update.
+	 */
+	$old_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s", $option ) );
+	if ( ! $old_autoload || $old_autoload === $autoload ) {
+		return false;
+	}
+
+	// Update the autoload value.
+	$result = $wpdb->update(
+		$wpdb->options,
+		array( 'autoload' => $autoload ),
+		array( 'option_name' => $option )
+	);
```

Comment:
I think this should result in a falsey value if the option name doesn't exist, becuase the `$wpdb->query()` for the update will return 0 for number of rows affected or `false` on an error. This should allow us to avoid the initial query to get the `$old_autoload` value.

Reply:
@joemcgill Good point. I would update this, however per your overarching comment, I'll make this function a wrapper of `wp_set_options_autoload()`, that way this code path won't be relevant anymore anyway.